src/generic/fetch.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Contains utility methods using 'fetch'.
+ */
+
+import {ApiError, ErrorCode} from '../core/api_error';
+import {BFSCallback} from '../core/file_system';
+
+/**
+ * Asynchronously download a file as a buffer or a JSON object.
+ * Note that the third function signature with a non-specialized type is
+ * invalid, but TypeScript requires it when you specialize string arguments to
+ * constants.
+ * @hidden
+ */
+export function fetchFileAsync(p: string, type: 'buffer', cb: BFSCallback<Buffer>): void;
+export function fetchFileAsync(p: string, type: 'json', cb: BFSCallback<any>): void;
+export function fetchFileAsync(p: string, type: string, cb: BFSCallback<any>): void;
+export function fetchFileAsync(p: string, type: string, cb: BFSCallback<any>): void {
+  fetch(p)
+    .then((res) => {
+      if (!res.ok) {
+        return cb(new ApiError(res.status, "fetch error."));
```

Comment:
Will changing the error code from status to EIO break any existing applications? I'm wondering if it's a breaking change.